### ISSUE TITLE
experiment(agent): iteratively optimize the Task 1 Q-learning agent

### ISSUE THREAD

**Issue by LiliWestermann** [1BlauNitrox/mle-final-project#35]
## Objective

Systematically improve the Task 1 performance of
`DerKleineVermoegensumverteiler` through a sequence of controlled experiments.

The campaign should produce a feature-based tabular Q-learning agent that
reliably collects visible coins in the `coin-heaven` scenario and is ready for
the final held-out evaluation defined in
`docs/0007-task-1-baseline-contract.md`.

This issue coordinates the optimization campaign. Each material design change
must be evaluated in a separate, prospectively defined experiment issue and
documented under `experiments/YYYY-MM-DD-short-name/`.

## Starting point

Use the agent revision merged from PR #32 as the initial baseline.

The current implementation uses:

- feature-based tabular Q-learning;
- an eight-element state representation;
- `alpha = 0.1`;
- `gamma = 0.9`;
- initial epsilon `1.0`;
- epsilon decay `0.99`;
- minimum epsilon `0.1`;
- rewards for `COIN_COLLECTED`, `INVALID_ACTION`, and `WAITED`;
- the Task 1 action set `UP`, `RIGHT`, `DOWN`, `LEFT`, `WAIT`.

The baseline performance must be measured before selecting modifications.

## Research questions

The campaign should investigate which controlled changes improve coin
collection and training reliability. Candidate experiment families include:

1. reward shaping;
2. epsilon schedule and exploration;
3. learning rate and discount factor;
4. training duration and convergence;
5. feature representation.

Each experiment must change one clearly defined factor or a justified,
pre-registered group of related factors.

## Experimental workflow

For every experiment:

1. state a falsifiable hypothesis;
2. identify the previous best agent revision as the baseline;
3. define the independent variable and controlled variables;
4. register training and development seeds before running;
5. use multiple independent training runs;
6. evaluate all variants under identical conditions;
7. report aggregate performance and variation across seeds;
8. preserve negative and mixed results;
9. select a new candidate only when the registered success criterion passes;
10. record the resulting decision and follow-up experiment.

Development evaluation must use only development seeds. Final held-out seeds
from `docs/0007-task-1-baseline-contract.md` must remain unopened until the
configuration is frozen.

## Primary metrics

Use the metrics required by the Task 1 baseline contract, including:

- coin-collection fraction;
- full-clear and zero-coin rates;
- coins collected;
- steps per collected coin;
- coins per 100 survival steps;
- invalid-action rate;
- episode length;
- cumulative shaped reward as a diagnostic metric;
- action counts;
- evaluation decision time.

Environment score alone is not sufficient for selecting a candidate.

## Experiment documentation

Each experiment must create:

```text
experiments/YYYY-MM-DD-short-name/
|-- README.md
|-- config.yaml
|-- summary.csv
`-- figures/
```

The README must contain:

- metadata and producing commit;
- hypothesis;
- baseline and tested variant;
- training and evaluation setup;
- complete seed lists;
- metrics and prospective success criterion;
- results and uncertainty;
- interpretation;
- decision and follow-up.

Raw logs, replay collections, and temporary checkpoints must remain outside
Git. Relevant final artifacts must be identified by SHA-256 checksum.

## Suggested experiment sequence

- [ ] Measure the unchanged PR #32 baseline on development seeds.
- [ ] Compare minimal rewards with a prospectively defined reward-shaping
      variant.
- [ ] Compare exploration schedules using the best reward configuration.
- [ ] Evaluate selected learning-rate and discount-factor configurations.
- [ ] Investigate feature changes only if the preceding diagnostics identify a
      representational limitation.
- [ ] Confirm the selected configuration with fresh development seeds.
- [ ] Freeze features, rewards, hyperparameters, evaluation code, and checkpoint
      selection.
- [ ] Run the final held-out Task 1 evaluation in a separate experiment.

The sequence may be adapted based on documented results. An adaptation must be
recorded before running the affected follow-up experiment.

## Campaign success criterion

The campaign is complete when:

- a selected candidate consistently outperforms the original PR #32 baseline on
  development seeds;
- improvements are supported by multi-seed evidence rather than a single best
  run;
- all experiments, including failed ones, are reproducibly documented;
- the final candidate and its configuration are frozen;
- the candidate is ready for the one-time held-out evaluation defined in
  `docs/0007-task-1-baseline-contract.md`.

Passing development experiments does not itself establish Task 1 completion.
That claim requires the separate final held-out evaluation.

## Acceptance criteria

- [ ] An owner and reviewer are assigned.
- [ ] The original PR #32 development baseline is recorded.
- [ ] Each material modification has its own experiment issue and hypothesis.
- [ ] Training, development, and final seed populations remain disjoint.
- [ ] Compared variants use identical evaluation conditions.
- [ ] Results include variation across training runs and world seeds.
- [ ] Negative and mixed results are retained.
- [ ] Experiment records contain configurations, data summaries, figures, and
      conclusions.
- [ ] The selected candidate is justified by development evidence.
- [ ] The final configuration is frozen before held-out evaluation.
- [ ] The agent card records the complete experiment lineage and limitations.
- [ ] Material AI assistance is reviewed and disclosed according to
      `docs/0006-ai-usage.md`.

## Out of scope

- bombs, crates, and opponents;
- Task 2-4 capabilities;
- DQN or another model family;
- repeated evaluation on the final held-out seeds;
- post-hoc removal of failed seeds or training runs;
- changing success thresholds after observing results;
- tournament-readiness claims.

**Comment by 1BlauNitrox** [comment]
The downstream freeze and cumulative-extension chain is now registered. #40 freezes the development-selected Task 1 tabular baseline after this campaign; #43 then prepares a behavior-preserving Task 2 successor; #45 implements cumulative Task 2 tabular capability. The frozen Task 1 directory must remain unchanged.
